### PR TITLE
Improve Documentation

### PR DESCRIPTION
**Pull request by naterarmstrong**
Improve the documentation (mostly code comments, but also including a new architecture method) pretty heavily, refactor some parts of the code to make it more legible.

### CODE REVIEW

**Comment by naterarmstrong** [comment]
There shouldn't be any functionality changes, so make sure to look over any areas which refactor to make sure I didn't miss anything, as there are quite a few changes.

File: doc/architecture.md [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+- a type for every storage item could be retrieved
+- an appropriate type for each extrinsic call that could be submitted
+
+`go-substrate-gen` uses [Jennifer](https://github.com/dave/jennifer) to perform the actual code generation. 
+
+## Overview of Code Structure
+The overall approach used is very simple:
+
+1. Parse the metadata returned from the `state.getMetadata` RPC endpoint by any substrate chain.
+1. Create a type generator which caches generated types
```
Comment (Aphoh) [inline doc/architecture.md]: 1,1,1?

File: doc/architecture.md [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+```json
+"pallets": [
+  {
+    "name": "pallet_name",
+    "storage": {...},
+    "calls": {...},
+    "events": {...},
+    "constants": {...},
+    "errors": {...},
+    "index": ...
```
Comment (Aphoh) [inline doc/architecture.md]: also use `{...}`

File: package.json [hunk 3]
```diff
@@ -0,0 +1 @@
+{}
```
Comment (Aphoh) [inline package.json]: Delete this file?

File: package-lock.json [hunk 4]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (Aphoh) [inline package-lock.json]: Delete this one too?

File: palletgen/storagegen/storagegen.go [hunk 5]
```diff
@@ -65,29 +67,52 @@ func (sg *StorageGenerator) GenPlain(v types.Si1LookupTypeID, item *types.Storag
...
 	}
 
-	args := []jen.Code{} // pointer to metadaa
-	var ind uint32 = 0
+	// Generate the arguments needed to specify a storage value within a map
+	args := []jen.Code{} // pointer to metadata
 	// Don't have a field name, so we prefix by the type name
+	var ind uint32 = 0
 	newArgs, keyArgNames, err := sg.tygen.GenerateArgs(gend, &ind, gend.DisplayName())
+	if err != nil {
```
Comment (Aphoh) [inline palletgen/storagegen/storagegen.go:91]: 🥲 good catch

**Review by Aphoh — APPROVED** [review]
Looks good, maybe do a quick regression test and then squash/merge.

File: doc/architecture.md (hunk 1 above)
Comment (elv-nate) [reply doc/architecture.md]: I guess that is a little ugly. Markdown numbers it properly anyway, if you look at the rendered version, but I guess I'll swap that over

**Pull request merged**